### PR TITLE
add main script experiments

### PR DESCRIPTION
**Pull request by tomsilver**
should we actually merge this into master? I don't see why not?

### CODE REVIEW

File: scripts/run_predicators_baseline_experiments.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        # Score function baselines.
+        python $FILE --experiment_id ${ENV}_prederror_${NUM_TRAIN_TASKS}demo --env $ENV --approach grammar_search_invention --grammar_search_score_function prediction_error --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+        python $FILE --experiment_id ${ENV}_branchfac_${NUM_TRAIN_TASKS}demo --env $ENV --approach grammar_search_invention --grammar_search_score_function branching_factor --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+        python $FILE --experiment_id ${ENV}_energy_${NUM_TRAIN_TASKS}demo --env $ENV --approach grammar_search_invention --grammar_search_score_function lmcut_energy_lookaheaddepth0 --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+
+        # Other baselines.
+        python $FILE --experiment_id ${ENV}_random --env $ENV --approach random_options --seed $SEED
+        python $FILE --experiment_id ${ENV}_gnn_shooting_${NUM_TRAIN_TASKS}demo --env $ENV --approach gnn_policy --seed $SEED
```
Comment (ronuchit) [inline scripts/run_predicators_baseline_experiments.sh]: add excluded predicates to all gnn approaches

File: scripts/run_predicators_baseline_experiments.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (ronuchit) [inline scripts/run_predicators_baseline_experiments.sh:1]: check chmod on both scripts

File: scripts/run_predicators_baseline_experiments.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    for ENV in ${ALL_ENVS[@]}; do
+        # Main approach.
+        python $FILE --experiment_id ${ENV}_main_${NUM_TRAIN_TASKS}demo --env $ENV --approach grammar_search_invention --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+
+        # Ablations.
+        # Note: downrefeval is main but with --sesame_max_skeletons_optimized 1 during evaluation only. We can only run this using `--load_approach` since we don't allow grammar_search_expected_nodes_max_skeletons to be greater than sesame_max_skeletons_optimized during invention.
+        python $FILE --experiment_id ${ENV}_downrefscore_${NUM_TRAIN_TASKS}demo --env $ENV --approach grammar_search_invention --grammar_search_expected_nodes_max_skeletons 1 --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+        python $FILE --experiment_id ${ENV}_noinventallexclude_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --excluded_predicates all --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
+        python $FILE --experiment_id ${ENV}_noinventnoexclude_${NUM_TRAIN_TASKS}demo --env $ENV --approach nsrt_learning --seed $SEED --num_train_tasks $NUM_TRAIN_TASKS
```
Comment (ronuchit) [inline scripts/run_predicators_baseline_experiments.sh:23]: let's rename all experiment IDs to match paper? it took me a while to remember that `noinventnoexclude` is Manual, etc

File: scripts/run_predicators_num_demos_experiments.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+FILE="scripts/submit.py"
+ALL_NUM_TRAIN_TASKS=(
+    "25"
+    "50"
+    "75"
+    "100"
+    "125"
+    "150"
+    "175"
+    "200"
```
Comment (ronuchit) [inline scripts/run_predicators_num_demos_experiments.sh:14]: we don't need 200 here

File: scripts/run_predicators_num_demos_experiments.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "cover"
+    "pybullet_blocks"
+    "painting"
+    "tools"
+)
+
+for SEED in $(seq $START_SEED $((NUM_SEEDS+START_SEED-1))); do
+    for ENV in ${ALL_ENVS[@]}; do
+        for NUM_TRAIN_TASKS in ${ALL_NUM_TRAIN_TASKS[@]}; do
+            # Main approach only.
```
Comment (ronuchit) [inline scripts/run_predicators_num_demos_experiments.sh:26]: should we do GNN here too?

File: scripts/run_predicators_baseline_experiments.sh (hunk 3 above)
Comment (tomsilver) [reply scripts/run_predicators_baseline_experiments.sh:23]: nah

**Review by ronuchit — APPROVED** [review]
okay hopefully this is good, only time will tell...

**Pull request merged**